data_juicer/ops/mapper/sentence_augmentation_mapper.py

```diff
@@ -0,0 +1,138 @@
+import torch
+
+from data_juicer.ops.base_op import OPERATORS, Mapper
+from data_juicer.utils.model_utils import get_model, prepare_model
+
+DEFAULT_SYSTEM_PROMPT = "A chat between a curious user and an artificial \
+    intelligence assistant. The assistant gives helpful, detailed, and \
+        polite answers to the user's questions."
+
+OP_NAME = 'sentence_augmentation_mapper'
+
+
+@OPERATORS.register_module(OP_NAME)
+class SentenceAugmentationMapper(Mapper):
+    """Mapper to optimize instruction.
+    Recommended model list: [
+        lmsys/vicuna-13b-v1.5
+        Qwen/Qwen2-7B-Instruct
+    ]
+    """
+    _accelerator = 'cuda'
+
+    def __init__(self,
+                 hf_model: str = 'Qwen/Qwen2-7B-Instruct',
+                 system_prompt: str = None,
+                 task_sentence: str = None,
+                 max_new_tokens=256,
+                 temperature=0.2,
+                 top_p=None,
+                 num_beams=1,
+                 *args,
+                 **kwargs):
+        """
+        Initialization method.
+        :param hf_model: Hugginface model id.
+        :param system_prompt: System prompt.
+        :param task_sentence: The instruction for the current task.
+        :param max_new_tokens: the maximum number of new tokens
+            generated by the model.
+        :param temperature: used to control the randomness of \
+            generated text. The higher the temperature, the more \
+                random and creative the generated text will be.
+        :param top_p: randomly select the next word from the group \
+            of words whose cumulative probability reaches p.
+        :param num_beams: the larger the beam search size, the higher \
+            the quality of the generated text.
+        :param args: extra args
+        :param kwargs: extra args
+        """
+        super().__init__(*args, num_proc=1, **kwargs)
+
+        if system_prompt is None:
+            system_prompt = DEFAULT_SYSTEM_PROMPT
+        self.system_prompt = system_prompt
+        self.hf_model = hf_model
+        self.max_new_tokens = max_new_tokens
+
+        self.model_key = prepare_model(model_type='huggingface',
+                                       pretrained_model_name_or_path=hf_model)
+        self.temperature = temperature
+        self.top_p = top_p
+        self.num_beams = num_beams
+        self.task_sentence = task_sentence
+
+    def process(self, sample=None, rank=None):
+
+        if 'vicuna' in self.hf_model:
```

Comment:
The only difference between different `hf_models` is the template for the input prompt.
 Could we create `input_prompt ` in branch while keeping the rest of the code shared?